src/components/home/Post.tsx

```diff
@@ -0,0 +1,177 @@
+  };
+
+  const nameStyle: SxStyleProp = {
+    padding: photo ? '0.5em' : '1em',
+    fontSize: 'medium',
+    bg: 'cardBg',
+    display: 'flex',
+    justifyContent: 'space-between',
+  };
+
+  const messageStyle: SxStyleProp = {
+    px: photo ? '0.3em' : '1em',
+    borderBottom: '0.3em solid',
+    borderColor: 'cardBg',
+    fontSize: 'small',
+    bg: 'cardBg',
+    maxHeight: [expand ? 'auto' : (photo ? '3.5em' : wrapperWidth-nameHeight), photo ? '3.5em' : wrapperWidth-nameHeight],
+    overflow: 'hidden',
+    position: 'relative',
+  };
+
+  const iconStyle: SxStyleProp = {
+    fontSize: 'larger',
+    display: ['none', 'block'],
+  };
+
+  const expandButtonStyle: SxStyleProp = {
+    bg: 'cardBg',
+    right: photo ? '0.3em' : '1em', 
+    bottom: 0,
+  };
+
+  //TODO: make the expanded view
+  const popup = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+
+  };
+
+  /**
+   * A resize observer is used to update the following dimension changes whenever the user
+   * resizes the browser/page:  
+   * - `wrapperWidth`, width of the main wrapper component – used for when making sure the entire post
+   * is a square or for making the image a square.  
+   * -  `nameHeight`, height of the component that displays the name – used for computing
+   * the available space left when constraining the post to a square.  
+   * - `messageHeight`, height of the component that displays the message – used for computing the 
+   * available space for displaying the image if the message is present.
+   */
+  useEffect(() => {
+    setWrapperWidth(wrapperRef.current.getBoundingClientRect().width);
+    const ro = new ResizeObserver((entries) => {
+      entries.forEach((entry: ResizeObserverEntry) => {
+        const element: Element = entry.target;
+
+        switch (element) {
+          case wrapperRef.current:
+            setWrapperWidth(entry.contentRect.width);
+            break;
+          case nameRef.current:
+            setNameHeight(element.getBoundingClientRect().height);
+            break;
+          case messageRef.current:
+            setMessageHeight(element.getBoundingClientRect().height);
+            if (element.scrollHeight > element.getBoundingClientRect().height) {
+              //TODO: fix this bug of overflow detection
+              console.log(element.scrollHeight, element.getBoundingClientRect().height);
+              setOverflow(true);
+            }
+            break;
+          default:
+            throw new Error('unregistered element');
+        }
+
+      });
+      
+    });
+
+    ro.observe(wrapperRef.current);
+    if (nameRef.current) ro.observe(nameRef.current);
+    if (messageRef.current) ro.observe(messageRef.current);
+    return () => ro.disconnect();
+  }, [nameRef.current, messageRef.current]);
+
+  
+  return (
+    <div sx={{...wrapperStyle, ...style}} ref={wrapperRef} onClick={popup}>
+      <div sx={nameStyle} ref={nameRef}>
+        {name}
```

Comment:
Does this account for anonymous names? Will it display nothing if no name, or will "Anonymous" be displayed?

Reply:
I was thinking it just displays nothing or do you think anonymous is better.

Reply:
Nothing works!